### DISCUSSION TITLE
Error installing seqret formula (M3 mac)

### DISCUSSION THREAD

**Discussion by AceM1188** [biostars/biostar-handbook-issues#335] [General]
Hi there,

I'm setting up a brand new bioinformatic computer for my lab with this (thank you btw). However, I'm running into an error specifically with installing the seqret formula. All other formula have installed without issue following the instructions provided.

Machine specs: Apple M3 Pro, OS: Sonoma 14.5. Xcode and OS are both up to date. 
Of note, anaconda has not been separately installed. 

I have tried both installing by the "Quick Installation" method as well as the "Step-by-Step" method. (I uninstalled everything using the code in the "Redo Installation" instructions prior to repeating using the Step-by-Step method). After both efforts, I receive the following error when trying to individually install seqret: 

`$ micromamba install seqret`
`bioconda/osx-64                                               No change`
`bioconda/noarch                                    @   2.3MB/s  2.3s`
`conda-forge/noarch                                  15.8MB @   3.5MB/s  4.5s`
`conda-forge/osx-64                                  32.2MB @   5.2MB/s  6.2s`

`Pinned packages:`
 ` - python 3.10.*`

`error    libmamba Could not solve for environment specs`
`    The following package could not be installed`
`    └─ seqret does not exist (perhaps a typo or a missing channel).`
`critical libmamba Could not solve for environment specs`

Of note, below are my channel priorities: 
`$ micromamba config list`
`channels:`
`  - conda-forge`
`  - bioconda`
`channel_priority: strict`

Any advice on how best to proceed to ensure a complete installation? Of course, please do not hesitate to let me know of any other info you need on my end. 

Thanks in advance.

**Comment by ialbert** [comment]
`seqret` is part of the emboss package,  so the installation would be

    micromamba install emboss

**Reply by AceM1188** [reply]
Also did this which did not generate any errors. Here's the output of `micromamba list`

`$ micromamba list`
`List of packages in environment: "/Users/mherna17/micromamba/envs/bioinfo"`

  `Name                       Version       Build                  Channel    `
`───────────────────────────────────────────────────────────────────────────────`
  `  agate                      1.12.0        pyh707e725_0           conda-forge`
 `  agate-dbf                  0.2.3         pyhd8ed1ab_0           conda-forge`
 `  agate-excel                0.2.3         py_0                   conda-forge`
 `  agate-sql                  0.7.2         pyhd8ed1ab_0           conda-forge`
`  aria2                      1.37.0        h21e4757_2             conda-forge`
`  babel                      2.14.0        pyhd8ed1ab_0           conda-forge`
`  bcftools                   1.20          h0126a07_1             bioconda   `
`  bedtools                   2.31.1        hdd9cfa2_2             bioconda   `
`  bioawk                     1.0           h10309d6_12            bioconda   `
`  biopython                  1.84          py310h936d840_0        conda-forge`
`  blast                      2.6.0         boost1.64_2            bioconda   `
`  bowtie2                    2.5.4         h8fd0fe2_2             bioconda   `
`  bwa                        0.7.18        h10309d6_1             bioconda   `
`  bzip2                      1.0.8         hfdf4475_7             conda-forge`
`  c-ares                     1.32.3        h51dda26_0             conda-forge`
`  ca-certificates            2024.7.4      h8857fd0_0             conda-forge`
`  cffi                       1.16.0        py310hdca579f_0        conda-forge`
`  cld2-cffi                  0.1.4         py310had63691_1005     conda-forge`
`  click                      8.1.7         unix_pyh707e725_0      conda-forge`
`  csvkit                     2.0.1         pyhd8ed1ab_0           conda-forge`
`  csvtk                      0.30.0        h527b516_2             bioconda   `
`  datamash                   1.8           h37ebe6b_0             conda-forge`
`  dbfread                    2.0.7         py_0                   conda-forge`
`  emboss                     6.5.7         2                      bioconda   `
`  et_xmlfile                 1.1.0         pyhd8ed1ab_0           conda-forge`
`  expat                      2.6.2         h73e2aa4_0             conda-forge`
`  fastp                      0.23.4        h1d47e6f_5             bioconda   `
`  fastqc                     0.12.1        hdfd78af_0             bioconda   `
`  font-ttf-dejavu-sans-mono  2.37          hab24e00_0             conda-forge`
`  fontconfig                 2.14.2        h5bb23bf_0             conda-forge`
`  freebayes                  1.3.6         hd0edb90_5             bioconda   `
`  freetype                   2.12.1        h60636b9_2             conda-forge`
`  future                     1.0.0         pyhd8ed1ab_0           conda-forge`
`  gawk                       5.3.0         h2c496e9_0             conda-forge`
`  gettext                    0.22.5        h5ff76d1_2             conda-forge`
`  gettext-tools              0.22.5        h5ff76d1_2             conda-forge`
`  gmp                        6.3.0         hf036a51_2             conda-forge`
`  greenlet                   3.0.3         py310h5daac23_0        conda-forge`
`  gsl                        2.7           h93259b0_0             conda-forge`
`  hisat2                     2.2.1         h7133b54_6             bioconda   `
`  htslib                     1.20          hec81eee_2             bioconda   `
`  icu                        75.1          h120a0e1_0             conda-forge`
`  isa-l                      2.31.0        hfdf4475_2             conda-forge`
`  isodate                    0.6.1         pyhd8ed1ab_0           conda-forge`
`  jq                         1.7.1         h10d778d_0             conda-forge`
`  krb5                       1.21.3        h37d8d59_0             conda-forge`
`  leather                    0.3.4         pyhd8ed1ab_0           conda-forge`
`  libasprintf                0.22.5        h5ff76d1_2             conda-forge`
`  libasprintf-devel          0.22.5        h5ff76d1_2             conda-forge`
`  libblas                    3.8.0         14_openblas            conda-forge`
`  libcblas                   3.8.0         14_openblas            conda-forge`
`  libcurl                    8.9.1         hfcf2730_0             conda-forge`
`  libcxx                     18.1.8        hef8daea_2             conda-forge`
`  libdeflate                 1.20          h49d49c5_0             conda-forge`
`  libedit                    3.1.20191231  h0678c8f_2             conda-forge`
`  libev                      4.33          h10d778d_2             conda-forge`
`  libexpat                   2.6.2         h73e2aa4_0             conda-forge`
`  libffi                     3.4.2         h0d85af4_5             conda-forge`
`  libgcc                     4.8.5         hdbeacc1_10            conda-forge`
`  libgettextpo               0.22.5        h5ff76d1_2             conda-forge`
`  libgettextpo-devel         0.22.5        h5ff76d1_2             conda-forge`
`  libgfortran                3.0.1         0                      conda-forge`
`  libhwloc                   2.11.1        default_h456cccd_1000  conda-forge`
`  libiconv                   1.17          hd75f5a5_2             conda-forge`
`  libidn2                    2.3.7         h10d778d_0             conda-forge`
`  libintl                    0.22.5        h5ff76d1_2             conda-forge`
`  libintl-devel              0.22.5        h5ff76d1_2             conda-forge`
`  liblapack                  3.8.0         14_openblas            conda-forge`
`  libnghttp2                 1.58.0        h64cf6d3_1             conda-forge`
`  libopenblas                0.3.7         hd44dcd8_1             conda-forge`
`  libpng                     1.6.43        h92b6c6a_0             conda-forge`
`  libsqlite                  3.46.0        h1b8f9f3_0             conda-forge`
`  libssh2                    1.11.0        hd019ec5_0             conda-forge`
`  libunistring               0.9.10        h0d85af4_0             conda-forge`
`  libxml2                    2.12.7        heaf3512_4             conda-forge`
`  libzlib                    1.3.1         h87427d6_1             conda-forge`
`  llvm-openmp                18.1.8        h15ab845_0             conda-forge`
`  loguru                     0.7.2         py310h2ec42d9_1        conda-forge`
`  mafft                      7.526         hfdf4475_0             conda-forge`
`  make                       4.3           h22f3db7_1             conda-forge`
`  minimap2                   2.1.1         0                      bioconda   `
`  mpfr                       4.2.1         hc80595b_2             conda-forge`
`  ncbi-datasets-cli          16.25.0       h694c41f_0             conda-forge`
`  ncurses                    6.5           h5846eda_0             conda-forge`
`  numpy                      1.22.4        py310hed37afb_0        conda-forge`
`  oniguruma                  6.9.9         h10d778d_0             conda-forge`
`  openjdk                    22.0.1        h1c1aa8c_1             conda-forge`
`  openpyxl                   3.1.2         py310h6729b98_1        conda-forge`
`  openssl                    3.3.1         h87427d6_2             conda-forge`
`  pandas                     2.2.2         py310hbf2a7f0_1        conda-forge`
`  parallel                   20240722      h694c41f_0             conda-forge`
`  parsedatetime              2.4           py_1                   conda-forge`
`  perl                       5.32.1        7_h10d778d_perl5       conda-forge`
`  perl-test-harness          3.44          pl5321hd8ed1ab_0       conda-forge`
`  perl-text-csv              2.02          pl5321hd8ed1ab_0       conda-forge`
`  pip                        24.2          pyhd8ed1ab_0           conda-forge`
`  polypolish                 0.6.0         h48ddd8c_1             bioconda   `
`  pycparser                  2.22          pyhd8ed1ab_0           conda-forge`
`  pyicu                      2.13.1        py310h3ba615c_1        conda-forge`
`  pypolca                    0.3.1         pyhdfd78af_0           bioconda   `
`  python                     3.10.14       h00d2728_0_cpython     conda-forge`
`  python-dateutil            2.9.0         pyhd8ed1ab_0           conda-forge`
`  python-slugify             8.0.4         pyhd8ed1ab_0           conda-forge`
`  python-tzdata              2024.1        pyhd8ed1ab_0           conda-forge`
`  python_abi                 3.10          4_cp310                conda-forge`
`  pytimeparse                1.1.8         py_0                   conda-forge`
`  pytz                       2024.1        pyhd8ed1ab_0           conda-forge`
`  pyyaml                     6.0.1         py310h6729b98_1        conda-forge`
`  readline                   8.2           h9e318b2_1             conda-forge`
`  samtools                   1.20          h94387ee_1             bioconda   `
`  seqkit                     2.8.2         h527b516_1             bioconda   `
`  seqtk                      1.2           0                      bioconda   `
`  setuptools                 71.0.4        pyhd8ed1ab_0           conda-forge`
`  six                        1.16.0        pyh6c4a22f_0           conda-forge`
`  snpeff                     5.0           hdfd78af_1             bioconda   `
`  sqlalchemy                 1.4.49        py310hb372a2b_1        conda-forge`
`  subread                    1.5.3         0                      bioconda   `
`  tabixpp                    1.1.2         he6e7409_3             bioconda   `
`  tbb                        2021.12.0     h3c5361c_3             conda-forge`
`  text-unidecode             1.3           pyhd8ed1ab_1           conda-forge`
`  tk                         8.6.13        h1abcd95_1             conda-forge`
`  trimmomatic                0.39          hdfd78af_2             bioconda   `
`  tzdata                     2024a         h0c530f3_0             conda-forge`
`  ucsc-bedgraphtobigwig      357           1                      bioconda   `
`  vcflib                     1.0.3         h0546b6b_4             bioconda   `
`  wget                       1.21.4        hca547e6_0             conda-forge`
`  wheel                      0.43.0        pyhd8ed1ab_1           conda-forge`
`  xlrd                       2.0.1         pyhd8ed1ab_3           conda-forge`
`  xz                         5.2.6         h775f41a_0             conda-forge`
`  yaml                       0.2.5         h0d85af4_2             conda-forge`
`  zlib                       1.3.1         h87427d6_1             conda-forge`
`  zstd                       1.5.6         h915ae27_0             conda-forge`

However, when trying to execute the `seqret` formula I get the following error: 
`$ seqret`
`dyld[94411]: Library not loaded: /usr/local/lib/libpq.5.dylib`
`  Referenced from: <879B2B8E-E83F-39A3-A92E-3E70DC2EA07C> /Users/mherna17/micromamba/envs/bioinfo/bin/_seqret`
`  Reason: tried: '/usr/local/lib/libpq.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/usr/local/lib/libpq.5.dylib' (no such file), '/usr/local/lib/libpq.5.dylib' (no such file), '/usr/lib/libpq.5.dylib' (no such file, not in dyld cache)`
`/Users/mherna17/micromamba/envs/bioinfo/bin/seqret: line 7: 94411 Abort trap: 6           _seqret $@`

And when I try `embossversion`:
`$ embossversion `
`dyld[94425]: Library not loaded: /usr/local/lib/libpq.5.dylib`
`  Referenced from: <AB00148E-B10B-3CE1-A3E0-DA813018CA45>` `/Users/mherna17/micromamba/envs/bioinfo/bin/_embossversion
  Reason: tried: '/usr/local/lib/libpq.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/usr/local/lib/libpq.5.dylib' (no such file), '/usr/local/lib/libpq.5.dylib' (no such file), '/usr/lib/libpq.5.dylib' (no such file, not in dyld cache)`
`/Users/mherna17/micromamba/envs/bioinfo/bin/embossversion: line 7: 94425 Abort trap: 6           _embossversion $@`

Any thoughts?

**Comment by ialbert** [comment]
not sure what that is but try installing into a separate env and run it that way

    micromamba create -n emboss emboss

then run via that environment

    micromamba run -n emboss seqret

**Reply by AceM1188** [reply]
Weird indeed. 
But installing and running via a separate environment worked!
Thank you!

**Comment by ialbert** [comment]
`seqret` has been removed from most content in the book. Most of what `seqre` could do can be achieved in simpler way with `bio fetch` etc.

The `emboss` dependency has now been removed from the installation. If someone needs tools from the Emboss suite  they should use the following construct:

```
# Install emboss into a separate environment.
micromamba create -n emboss emboss


# When an emboss tool is needed, run it in the emboss environment.
micromamba run -n emboss seqret
```
